### ISSUE TITLE
Support for built-in Node.js FormData

### ISSUE THREAD

**Issue by chrisdickinson** [fastify/light-my-request#298]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

Hi folks! Thanks for the great work on Fastify.

Piggybacking of off https://github.com/fastify/light-my-request/issues/35 – now that there's a current Node.js release (v22) that supports [`FormData`](https://nodejs.org/api/globals.html#class-formdata) as a builtin, could this library support it directly? The existing solution, `form-data`, seems to include some deprecated code and I suspect the difference between the builtin and the package-provided `FormData` might be hard to parse for folks new to the ecosystem.

I understand that there might be some difficulties with supporting native `FormData`. In particular, if I understand correctly, there's a "special relation" between `RequestInit` and `FormData` that gives the former access to the latter's headers; so in a `fetch('/url', {body: formData})` there's no need to set the headers specifically. The `form-data` package supports accessing these headers through a `.getHeaders()` method, but I don't know of anything similar for builtin `FormData`.

I'm asking this here because of the fastify team's collective experience building the underpinnings of Node's `fetch` implementation in undici; if you have some pointers as to how to access the headers of builtin `FormData`, I'd be happy to take on this work. (Alternatively, if you say "this will be quite a yak shave", at least this issue can warn folks about the presence of yaks on the trail :smile:)

**Comment by climba03003** [comment]
I don't understand what you means.
Doesn't it already support `FormData` as body?
